code/modules/projectiles/laser.dm

```diff
@@ -498,6 +498,31 @@ toxic - poisons
 				V.health -= power * 1.8
 				V.checkhealth()
 
+	sawfly// projectile of choice for sawflies
+		name = "Whirring Blades"
+
+		damage_type = D_PIERCING
+		damtype = D_PIERCING // weirdly enough, giving them stabbing damage goes through armor better than peircing
+		shot_sound = null
+		hit_human_sound = "sound/machines/chainsaw_green.ogg"
+		power = 16
+		dissipation_rate = 16
+		sname = "Whirring Blades"
+
+
+		on_pre_hit(var/atom/hit, var/angle, var/obj/projectile/O)
+			var/mob/M = hit
+			if (istraitor(M) || isnukeop(M) || isspythief(M)) // uh oh! we just hit a friend
```

Comment:
Below you do the same check but including `|| (M in src.friends)`, you can do the same here by getting `O.mobshooter.friends` (with appropriate casting)